src/client/datascience/editor-integration/codewatcher.ts

```diff
@@ -1052,10 +1092,11 @@ export class CodeWatcher implements ICodeWatcher {
             if (this.document) {
                 // Use that to get our code.
                 const code = this.document.getText(currentRunCellLens.range);
-                await this.addCode(
+                this.queueAddCode(
```

Comment:
Isn't this the place where we need to chain the promises, the way Rich suggested.
Basically just chaning the promises and if it fails, then clear the promise (from property).
And I think this method should not  return a promise, instead it should be sync that way calling code will never await.

Reply:
I think we need to await because it checks the result? It then does something. Not sure what addError does.